esadapter/src/main/java/com/ncc/neon/adapters/es/ElasticsearchAdapter.java

```diff
@@ -459,4 +455,47 @@ public void onFailure(Exception e) {
             });
         });
     }
+
+    @Override
+    public Mono<ActionResult> deleteData(MutateQuery mutateQuery) {
+        DeleteRequest deleteRequest = ElasticsearchQueryConverter.convertMutationDeleteQuery(mutateQuery);
+        return Mono.create(sink -> {
+            client.deleteAsync(deleteRequest, RequestOptions.DEFAULT, new ActionListener<DeleteResponse>() {
+                @Override
+                public void onResponse(DeleteResponse deleteResponse) {
+                    String deleteText = "";
+                    boolean deleteFailed = false;
+                    switch (deleteResponse.getResult()) {
+                        case CREATED:
+                            deleteText = "created successfully";
+                            break;
+                        case UPDATED:
+                            deleteText = "updated successfully";
+                            break;
+                        case DELETED:
+                            deleteText = "deleted successfully";
+                            break;
+                        case NOOP:
+                            deleteText = "no operation needed";
+                            break;
+                        case NOT_FOUND:
+                            deleteText = "not found";
+                            deleteFailed = true;
+                            break;
+                    }
+                    String responseText = "Index " + deleteResponse.getIndex() + " ID " + deleteResponse.getId() +
+                            " " + deleteText + ".";
+                    List<String> documentErrors = deleteFailed ? new ArrayList<String>() {{
+                        add(responseText);
+                    }} : new ArrayList<String>();
```

Comment:
Could we extract this behavior into a separate shared method, since I think it's almost the same between Delete, Insert, and Mutate?  I'm assuming the DeleteResponse, IndexResponse, and UpdateResponse share a common superclass that you could use?

Reply:
Fixed!